### DISCUSSION TITLE
Black screen after selecting boot drive

### DISCUSSION THREAD

**Discussion by controlfix87** [osy86/HaC-Mini#918] [Q&A] [answered]
Hi,
I've updated my kexts in the EFI partition, now the system is showing black screen when booting into MacOS AFTER I see the apple logo loading (when user password is required)
How can I restore the original kexts files? I have it on USB drive I've created before...
Thanks

**Comment by scottzoloft** [comment]
You probably don't want to update your kexts manually. Some kexts may not work out of the box. Wait for a new release of the hac mini installer for your kexts unless you want to test ahead. If you haven't gotten very far i'd just start the mac os installer over and wipe your drive. Make a new EFI from the installer

**Reply by scottzoloft** [reply]
Can also try booting from the USB installer and get back into your mac os partition and try restoring your EFI that way. Re-run the installer and point it at your Mac OS partition.

**Reply by controlfix87** [reply]
I can’t click on mark as answer.. any ideas?..

**Reply by controlfix87** [reply]
How can I run the pkg installer from the usb stick macos installer? 
am I able to launch pkg files there?

**Reply by scottzoloft** [reply]
> How can I run the pkg installer from the usb stick macos installer?
> am I able to launch pkg files there?

what i meant was once you get back into Mac OS, re-download the installer from the project page, re-run, follow the instructions and point the installer at your local mac OS drive with whatever advanced options you'd like, and that process should generate a new EFI folder with your boot EFIs, kexts, boot args, and serial info on your local OS drive's EFI partition. That way you won't don't have to boot off USB to get into Mac OS. you may have already done this though

**Reply by controlfix87** [reply]
> > How can I run the pkg installer from the usb stick macos installer?
> > am I able to launch pkg files there?
> 
> what i meant was once you get back into Mac OS, re-download the installer from the project page, re-run, follow the instructions and point the installer at your local mac OS drive with whatever advanced options you'd like, and that process should generate a new EFI folder with your boot EFIs, kexts, boot args, and serial info on your local OS drive's EFI partition. That way you won't don't have to boot off USB to get into Mac OS. you may have already done this though

The main issue was to restore EFI without the need to reinstall thw whole system again… :)

**Comment by controlfix87** [comment, marked as answer]
Thanks for the reply!
I’ve booted from Gparted usb stick and copied the original EFI folder to my system EFI and now it works as it should be.. lesson learned! :)
